src/vs/workbench/electron-browser/main.contribution.ts

```diff
@@ -104,6 +104,16 @@ let workbenchProperties: { [path: string]: IJSONSchema; } = {
 		'description': nls.localize('showEditorTabs', "Controls if opened editors should show in tabs or not."),
 		'default': true
 	},
+	'workbench.editor.tabDescription': {
+		'type': 'string',
+		'default': '${pathDifferential}',
+		'description': nls.localize({ comment: ['This is the description for a setting. Values surrounded by parenthesis are not to be translated.'], key: 'tabDescription' },
+			`Controls tab descriptions. Variables are substituted based on the context:
+\${pathShort}: e.g. parent
```

Comment:
This seems to be a very specific way of treating description labels by using the native path separator. You have to keep in mind though that there are editors that are not about files (e.g. extension editor). I would rather call this:
* `descriptionShort`
* `descriptionMedium`
* `descriptionLong`
* `descriptionDifferential`

Reply:
Do note that descriptions are already assumed to be paths when they're trimmed via `shorten`. That said, the architecture you mention below sounds sufficiently flexible and could be augmented to provide an editor-type-specific `shorten` method as well. Unfortunately, that's more involved than I can commit to.

Reply:
Yes, I agree that the shorten logic is assuming paths but that is only because this algorithm can only work with paths and nothing else, so I think that is fair. But other editors might have the desire to provide short, medium and long descriptions. 

I think the change is straightforward: I am not asking to implement this for each and every editor, it is fine to just move your logic into `FileEditorInput`.